src/Datadog.Trace/Logging/DatadogSerilogLogger.cs

```diff
@@ -0,0 +1,210 @@
+            => Write(LogEventLevel.Error, exception, messageTemplate, property0, property1, property2, sourceLine, sourceFile);
+
+        public void Error(Exception exception, string messageTemplate, object[] args, [CallerLineNumber] int sourceLine = 0, [CallerFilePath] string sourceFile = "")
+            => Write(LogEventLevel.Error, exception, messageTemplate, args, sourceLine, sourceFile);
+
+        private void Write<T>(LogEventLevel level, Exception exception, string messageTemplate, T property, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write<T0, T1>(LogEventLevel level, Exception exception, string messageTemplate, T0 property0, T1 property1, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property0, property1 }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write<T0, T1, T2>(LogEventLevel level, Exception exception, string messageTemplate, T0 property0, T1 property1, T2 property2, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property0, property1, property2 }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write(LogEventLevel level, Exception exception, string messageTemplate, object[] args, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid rate limiting calculation if log level is disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, args, sourceLine, sourceFile);
+            }
+        }
+
+        private void WriteIfNotRateLimited(LogEventLevel level, Exception exception, string messageTemplate, object[] args, int sourceLine, string sourceFile)
+        {
+            try
+            {
+                if (_rateLimiter.ShouldLog(sourceFile, sourceLine, out var skipCount))
+                {
+                    var newArgs = new object[args.Length + 1];
+                    Array.Copy(args, newArgs, args.Length);
+                    newArgs[args.Length] = skipCount;
+
+                    _logger.Write(level, exception, messageTemplate + ", {SkipCount} additional messages skipped", newArgs);
```

Comment:
Shouldn't we add the `additional messages skipped` bit only when SkipCount > 0 ? Otherwise I'm afraid it'll just add noise to the logs

Reply:
As I understand it from the RFC, no, it should always be there. But that's not explicitly mentioned, and I'm inclined to agree 👍!

Reply:
Changed it in https://github.com/DataDog/dd-trace-dotnet/pull/1153/commits/05de3984613bd001f05f2c5f22b34db3ccb4d2e5 as makes sense I think